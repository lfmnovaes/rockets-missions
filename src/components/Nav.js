import React from 'react';
import { NavLink } from 'react-router-dom';

const Nav = () => (
  <nav>
    <h1 className="navbar-brand">Space Traveler's Hub</h1>
    <ul className="navbar-nav">
      <li className="nav-item">
        <NavLink to="/">Rockets</NavLink>
      </li>
      <li className="nav-item">
        <NavLink to="/missions">Missions</NavLink>
      </li>
      <li className="nav-item">
        <NavLink to="/profile">My Profile</NavLink>
      </li>
    </ul>
  </nav>
);

export default Nav;
