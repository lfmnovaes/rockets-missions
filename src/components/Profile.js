import React from 'react';
import { Container, ListGroup } from 'react-bootstrap';
import { useSelector } from 'react-redux';

export default function Profile() {
  const missionsStore = useSelector((store) => store.missionsReducer);
  const reservedMissions = missionsStore.filter((mission) => mission.reserved === true);

  const rocketsStore = useSelector((store) => store.rocketsReducer);
  const reservedRockets = rocketsStore.filter((rocket) => rocket.reserved === true);

  return (
    <Container className="d-flex justify-content-evenly">
      <ListGroup>
        <h2>My Rockets</h2>
        {reservedRockets.map((rocket) => (
          <ListGroup.Item key={rocket.id}>{rocket.rocket_name}</ListGroup.Item>
        ))}
      </ListGroup>
      <ListGroup>
        <h2>My missions</h2>
        {reservedMissions.map((mission) => (
          <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
        ))}
      </ListGroup>
    </Container>
  );
}
