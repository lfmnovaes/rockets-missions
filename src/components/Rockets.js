import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  getRockets,
  AddReserve,
  CancelReserve,
} from '../redux/rockets/rockets';
import '../App.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const rocketList = useSelector((data) => data.rocketsReducer);

  useEffect(() => {
    if (rocketList.length === 0) {
      dispatch(getRockets());
    }
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const ReserveHandler = (id) => {
    dispatch(AddReserve(id));
  };

  const CancelReserveHandler = (id) => {
    dispatch(CancelReserve(id));
  };

  const rocketItem = rocketList.map((item) => (
    <div key={item.id} className="rocket-row">
      <img src={item.flickr_images[0]} alt={item.rocket_name} />
      <div className="rocket-detail">
        <h3>{item.rocket_name}</h3>
        <div className="description">
          {item.reserved ? <span>Reserved</span> : null}
          <p>{item.description}</p>
        </div>
        {item.reserved ? (
          <button
            type="submit"
            onClick={() => CancelReserveHandler(item.id)}
            className="cancelButton"
          >
            Cancel Reservation
          </button>
        ) : (
          <button
            type="submit"
            onClick={() => ReserveHandler(item.id)}
            className="reserveButton"
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  ));
  return <div className="rockets-list">{rocketItem}</div>;
};

export default Rockets;
