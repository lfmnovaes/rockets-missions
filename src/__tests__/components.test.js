import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import reducerMissions from '../redux/missions/missions';
import Rockets from '../components/Rockets';
import Missions from '../components/MissionList';
import Profile from '../components/Profile';
import Nav from '../components/Nav';

describe('Group of tests for Rockets page', () => {
  test('renders Rockets component correctly', () => {
    const rockets = renderer
      .create(<Provider store={store}><Rockets /></Provider>)
      .toJSON();
    expect(rockets).toMatchSnapshot();
  });
  test('List is created', () => {
    const { container } = render(<Provider store={store}><Rockets /></Provider>);
    const list = container.querySelector('.rockets-list');
    expect(list).toBeInTheDocument();
  });
  test('renders Profile Rockets correctly', () => {
    const rockets = renderer
      .create(<Provider store={store}><Profile /></Provider>)
      .toJSON();
    expect(rockets).toMatchSnapshot();
  });
});

describe('Group of tests for Missions page', () => {
  test('Test if Missions page is rendering correctly', () => {
    const missions = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(missions).toMatchSnapshot();
  });
  test('Check the initialState of the reducer', () => {
    expect(reducerMissions(undefined, {})).toEqual([]);
  });
});

describe('Group of tests for Profile page', () => {
  test('Test if Profile page is rendering correctly', () => {
    const profile = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(profile).toMatchSnapshot();
  });
});

describe('Test Nav renderisation', () => {
  test('Nav', () => {
    const nav = renderer
      .create(<BrowserRouter><Nav /></BrowserRouter>)
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
