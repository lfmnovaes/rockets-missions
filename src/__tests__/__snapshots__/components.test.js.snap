// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group of tests for Missions page Test if Missions page is rendering correctly 1`] = `
<table
  className="table table-sm table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Name
      </th>
      <th
        scope="col"
      >
        Description
      </th>
      <th
        scope="col"
      >
        Status
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;

exports[`Group of tests for Profile page Test if Profile page is rendering correctly 1`] = `
<div
  className="d-flex justify-content-evenly container"
>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <h2>
      My Rockets
    </h2>
  </div>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <h2>
      My missions
    </h2>
  </div>
</div>
`;

exports[`Group of tests for Rockets page renders Profile Rockets correctly 1`] = `
<div
  className="d-flex justify-content-evenly container"
>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <h2>
      My Rockets
    </h2>
  </div>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <h2>
      My missions
    </h2>
  </div>
</div>
`;

exports[`Group of tests for Rockets page renders Rockets component correctly 1`] = `
<div
  className="rockets-list"
/>
`;

exports[`Test Nav renderisation Nav 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container-fluid"
  >
    <span
      className="navbar-brand"
    >
      <img
        alt="Space Travelers logo"
        className="d-inline-block align-top"
        height="40"
        src="planet.png"
        width="40"
      />
       
      Space Travelers' Hub
    </span>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="divider-vertical"
      />
      <li
        className="nav-item profile"
      >
        <a
          className="nav-link"
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
